lib/Dialect/TritonGPU/IR/LinearLayoutConversions.cpp

```diff
@@ -1460,31 +1451,16 @@ LinearLayout chooseScaledMfmaScaleLayout(
   StringAttr kLane = StringAttr::get(ctx, "lane");
   StringAttr kWarp = StringAttr::get(ctx, "warp");
   StringAttr kBlock = StringAttr::get(ctx, "block");
-  // Init register layout. Will be adjusted later
-  auto regs = mlir::triton::identityStandardND(kRegister, {1, 1}, order);
-  LinearLayout lanes = LinearLayout::empty();
+  auto kDim = dotOperandIdx == 0 ? rank - 1 : rank - 2;
+  auto tilePerWarpNonK = tilesPerWarp[kDim];
```

Comment:
here we are dealing with operands and there is kDim and nonKDim. If you need nonKDim, you should use tilesPerWarp[nonKDim]